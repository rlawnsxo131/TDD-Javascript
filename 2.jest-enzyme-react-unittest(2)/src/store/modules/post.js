import { handleActions, createAction } from 'redux-actions';
import axios from 'axios';

function getPostAPI(postId) {
  return axios.get(`http://jsonplaceholder.typicode.com/posts/${postId}`);
}

const GET_POST_PENDING = 'post/GET_POST_PENDING';
const GET_POST_SUCCESS = 'post/GET_POST_SUCCESS';
const GET_POST_FAILURE = 'post/GET_POST_FAILURE';

const getPostPending = createAction(GET_POST_PENDING);
const getPostSuccess = createAction(GET_POST_SUCCESS);
const getPostFailure = createAction(GET_POST_FAILURE);

export const getPost = (postId) => async (dispatch) => {
  dispatch(getPostPending());
  try {
    const response = await getPostAPI(postId);
    dispatch(getPostSuccess(response));
  } catch (e) {
    dispatch(getPostFailure(e));
  }
}

const initialState = {
  fetching: false,
  error: false,
  title: '',
  body: '',
};

export default handleActions({
  [GET_POST_PENDING]: (state, action) => ({ ...state, fetching: true, error: false }),
  [GET_POST_SUCCESS]: (state, action) => ({
    ...state,
    fetching: false,
    title: action.payload.data.title,
    body: action.payload.data.body
  }),
  [GET_POST_FAILURE]: (state, action) => ({ ...state, fetching: false, error: true })
}, initialState);
