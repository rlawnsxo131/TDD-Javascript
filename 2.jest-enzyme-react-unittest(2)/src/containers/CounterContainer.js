import React, { Component } from 'react';
import Counter from '../components/Counter';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as counterActions from '../store/modules/counter';

class CounterContainer extends Component {
  onIncrease = () => {
    const { CounterActions } = this.props;
    CounterActions.increase();
    
  }
  onDecrease = () => {
    const { CounterActions } = this.props;
    CounterActions.decrease();
  }
  render() {
    const { number } = this.props;
    const { onIncrease, onDecrease } = this;
    return (
      <Counter
        value={number}
        onIncrease={onIncrease}
        onDecrease={onDecrease}
      />
    );
  }
}

export default connect(
  (state) => ({
    number: state.counter.number
  }),
  (dispatch) => ({
    CounterActions: bindActionCreators(counterActions, dispatch)
  })
)(CounterContainer);