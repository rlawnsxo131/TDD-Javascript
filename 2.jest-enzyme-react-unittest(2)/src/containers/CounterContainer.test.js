import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import CounterContainer from './CounterContainer';
import configureMockStore from 'redux-mock-store';
import * as counterActions from '../store/modules/counter';

describe('CounterContainer', () => {
  let component = null;
  let buttons = null;
  const mockStore = configureMockStore();

  // 데이터들을 받아올 가짜 스토어 만들기
  let store = mockStore({
    counter: {
      number: 0
    }
  });

  it('renders properly', () => {
    component = mount(
      <Provider store={store}>
        <CounterContainer />
      </Provider>
    );
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('dispatches INCREASE action', () => {
    component.find('button').at(0).simulate('click');
    expect(store.getActions()[0]).toEqual(counterActions.increase());
  });

  it('dispatches DECREASE action', () => {
    component.find('button').at(1).simulate('click');
    expect(store.getActions()[1]).toEqual(counterActions.decrease());
  });  
});
