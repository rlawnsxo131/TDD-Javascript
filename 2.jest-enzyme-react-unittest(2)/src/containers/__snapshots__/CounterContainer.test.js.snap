// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterContainer matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CounterContainer)>
    <CounterContainer
      CounterActions={
        Object {
          "decrease": [Function],
          "default": [Function],
          "increase": [Function],
        }
      }
      number={0}
    >
      <Counter
        onDecrease={[Function]}
        onIncrease={[Function]}
        value={0}
      >
        <div>
          <h1>
            카운터
          </h1>
          <h2>
            0
          </h2>
          <button
            onClick={[Function]}
          >
            +
          </button>
          <button
            onClick={[Function]}
          >
            -
          </button>
        </div>
      </Counter>
    </CounterContainer>
  </Connect(CounterContainer)>
</Provider>
`;
